src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +561,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
```

Comment:
>If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then [](start = 12, length = 88)

It feels to me that we are unnecessary forking the code and duplicating portion of the logic in the ```Else```.

Reply:
> That was the same comment Julien had but every attempt to change it caused an existing test to fail

That doesn't help with the review, since we don't know which tests are failing :)

I agree that it seems very weird how much duplicaiton there is.  I want to understand why the logic can't simply be:

1. consume whitespace if present.
2. consume '_' if present.
3. consume whitespace if present.
4. consume comment if present.
5. consume newline if present.

If this logic is breaking something, i want to understand *that case*, not just that some test failed :)  
Tanks!

Reply:
> This has the advantage that we didn't have to retest everything 

I am not sure we can say that because there are code changes on all branches. So far, I do not see an advantage having this split and we should try to eliminate it, I think. 

---
In reply to: [213422753](https://github.com/dotnet/roslyn/pull/28327#discussion_r213422753) [](ancestors = 213422753)

Reply:
I will try again, the case that would fail is _ whitespace something other than ' or EOF. This is a BadToken Error handled by higher level code.

1. consume whitespace if present. - always safe
2. consume '_' if present. - Can't do this in the case of most scan failures If you return false higher levels expect to attach error to _
3. consume whitespace if present. - can't do this if there is something after the space that is not _ or ' 
4. consume comment if present. - always safe
5. consume newline if present. - always safe

Reply:
Rewritten to address this, needs review.